plugin-server/src/worker/ingestion/persons/repositories/postgres-person-repository.test.ts

```diff
@@ -1128,6 +1135,512 @@ describe('PostgresPersonRepository', () => {
+            })
+        })
+
+        describe('PersonPropertiesSizeViolationError', () => {
+            it('should create error with correct properties', () => {
+                const error = new PersonPropertiesSizeViolationError('Test message', 123, 'person-id', 'distinct-id')
+
+                expect(error.message).toBe('Test message')
+                expect(error.teamId).toBe(123)
+                expect(error.personId).toBe('person-id')
+                expect(error.distinctId).toBe('distinct-id')
+                expect(error.name).toBe('PersonPropertiesSizeViolationError')
+            })
+        })
+
+        describe('createPerson with oversized properties', () => {
+            it('should throw PersonPropertiesSizeViolationError when properties exceed size limit', async () => {
+                const team = await getFirstTeam(hub)
+                const uuid = new UUIDT().toString()
+                const oversizedProperties = {
+                    description: 'x'.repeat(200),
+                }
+
+                const originalQuery = postgres.query.bind(postgres)
+                const mockQuery = jest.spyOn(postgres, 'query').mockImplementation(async (use, query, values, tag) => {
+                    if (typeof query === 'string' && query.includes('INSERT INTO posthog_person')) {
+                        const error = new Error('Check constraint violation')
+                        ;(error as any).code = '23514'
+                        ;(error as any).constraint = 'check_properties_size'
+                        throw error
+                    }
+                    return originalQuery(use, query, values, tag)
+                })
+
+                await expect(
+                    oversizedRepository.createPerson(
+                        TIMESTAMP,
+                        oversizedProperties,
+                        {},
+                        {},
+                        team.id,
+                        null,
+                        true,
+                        uuid,
+                        [{ distinctId: 'test-oversized' }]
+                    )
+                ).rejects.toThrow(PersonPropertiesSizeViolationError)
+
+                await expect(
+                    oversizedRepository.createPerson(
+                        TIMESTAMP,
+                        oversizedProperties,
+                        {},
+                        {},
+                        team.id,
+                        null,
+                        true,
+                        uuid,
+                        [{ distinctId: 'test-oversized-2' }]
+                    )
+                ).rejects.toThrow('Person properties create would exceed size limit')
+
+                mockQuery.mockRestore()
+            })
+        })
+
+        describe('updatePerson with oversized properties', () => {
+            it('should trim existing oversized person properties and update successfully', async () => {
```

Comment:
suggestion: Would be great to have one more test at the `updatePerson` method level for updating a person whose stored properties can't be trimmed enough, i.e. the protected properties stored in the DB before the update exceed the limit. 

There's potential for infinite loops in this scenario, so it would be great to have a test that covers this edge case.